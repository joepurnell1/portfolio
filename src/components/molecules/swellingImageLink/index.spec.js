import React from 'react';
import renderer from 'react-test-renderer';
import styled from 'styled-components';
import SwellingLink from './index';

const Container = styled('div')``;

describe('link component', () => {
  it('should render default as expected', () => {
    const render = renderer.create(
      <SwellingLink
        link="https://open.spotify.com/album/63McpWDS7P0dF06fqXQMCZ"
        imageSource="https://upload.wikimedia.org/wikipedia/en/7/70/Childish-gambino-because-the-internet.gif"
        imageAlt="becausetheinternet"
        size="50px"
        hoverSize="55px"
      />,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });

  it('should render swell on parent as expected', () => {
    const render = renderer.create(
      <Container>
        <SwellingLink
          swellOn={Container}
          link="https://open.spotify.com/album/63McpWDS7P0dF06fqXQMCZ"
          imageSource="https://upload.wikimedia.org/wikipedia/en/7/70/Childish-gambino-because-the-internet.gif"
          imageAlt="becausetheinternet"
          size="50px"
          hoverSize="55px"
        />
      </Container>,
    );
    expect(render.toJSON()).toMatchSnapshot();
  });
});
