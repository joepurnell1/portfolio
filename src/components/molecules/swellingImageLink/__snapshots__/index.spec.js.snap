// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`swelling image link component should render default as expected 1`] = `
<a
  className="sc-bdVaJa jrSDnk"
  href="https://open.spotify.com/album/63McpWDS7P0dF06fqXQMCZ"
  size="55px"
>
  <img
    alt="becausetheinternet"
    className="sc-bwzfXH eloVrX"
    height="50px"
    src="https://upload.wikimedia.org/wikipedia/en/7/70/Childish-gambino-because-the-internet.gif"
    width="50px"
  />
</a>
`;

exports[`swelling image link component should render swell on parent as expected 1`] = `
<div
  className="sc-ifAKCX cFlEyZ"
>
  <a
    className="sc-bdVaJa jrSDnk"
    href="https://open.spotify.com/album/63McpWDS7P0dF06fqXQMCZ"
    size="55px"
  >
    <img
      alt="becausetheinternet"
      className="sc-htpNat cpskcg"
      height="50px"
      src="https://upload.wikimedia.org/wikipedia/en/7/70/Childish-gambino-because-the-internet.gif"
      width="50px"
    />
  </a>
</div>
`;
